### PR TITLE
IDEX-2842 Move Docker client and runner from plugins to che-plugins

### PR DESCRIPTION
**Pull request by benoitf**

### CODE REVIEW

**Comment by garagatyi** [comment]
For docker API client:
you've created package org.eclipse.che.docker
whereas in 4.0 package is org.eclipse.che.plugin.docker

can you change your package to make master and 4.0 synchronous

The same for docker runner and che-plugin-docker-ext-client.

**Comment by benoitf** [comment]
ok, refactoring done

File: plugin-docker/che-plugin-docker-client/src/main/resources/org/eclipse/che/plugin/docker/client/dto/Dto.gwt.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (garagatyi) [inline plugin-docker/che-plugin-docker-client/src/main/resources/org/eclipse/che/plugin/docker/client/dto/Dto.gwt.xml]: Why you need client side gwt file in docker API client?

File: plugin-docker/che-plugin-docker-client/pom.xml [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bcpkix-jdk15on</artifactId>
+            <version>${org.bouncycastle.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.che.core</groupId>
+            <artifactId>che-core-api-core</artifactId>
+            <version>${che.core.version}</version>
+        </dependency>
+        <dependency>
```
Comment (garagatyi) [inline plugin-docker/che-plugin-docker-client/pom.xml]: why we need gwt here?

File: plugin-docker/pom.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    <artifactId>che-plugin-docker-parent</artifactId>
+    <packaging>pom</packaging>
+    <name>Che Plugin :: Docker :: Parent</name>
+    <modules>
+        <module>che-plugin-docker-client</module>
+        <module>che-plugin-docker-runner</module>
+        <module>che-plugin-docker-ext-client</module>
+    </modules>
+    <properties>
+        <dto-generator-out-directory>${project.build.directory}/generated-sources/dto/</dto-generator-out-directory>
```
Comment (garagatyi) [inline plugin-docker/pom.xml]: why we need dto here?

File: plugin-docker/pom.xml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        <gwt.test.skip>true</gwt.test.skip>
+    </properties>
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.eclipse.che.core</groupId>
```
Comment (garagatyi) [inline plugin-docker/pom.xml]: why we need dto maven plugin here?

**Comment by benoitf** [comment]
So, to sum up quickly.
There are DTO because the ext-client is using these DTO
And Client DTO are requiring GWT

**Comment by skabashnyuk** [comment]
ok

**Pull request merged**